webviz_subsurface/plugins/_pvt_plot/_plugin.py

```diff
@@ -0,0 +1,199 @@
+
+        self.ensemble_paths = {
+            ensemble: webviz_settings.shared_settings["scratch_ensembles"][ensemble]
+            for ensemble in ensembles
+        }
+
+        self.plotly_theme = webviz_settings.theme.plotly_theme
+
+        self.pvt_relative_file_path = pvt_relative_file_path
+
+        self.read_from_init_file = read_from_init_file
+
+        self.drop_ensemble_duplicates = drop_ensemble_duplicates
+
+        if self.pvt_relative_file_path is None:
+            self.pvt_df = load_pvt_dataframe(
+                self.ensemble_paths,
+                use_init_file=read_from_init_file,
+                drop_ensemble_duplicates=drop_ensemble_duplicates,
+            )
+        else:
+            # Load data from all ensembles into a pandas DataFrame
+            self.pvt_df = load_pvt_csv(
+                ensemble_paths=self.ensemble_paths,
+                csv_file=self.pvt_relative_file_path,
+                drop_ensemble_duplicates=drop_ensemble_duplicates,
+            )
+
+            self.pvt_df = self.pvt_df.rename(str.upper, axis="columns").rename(
+                columns={"TYPE": "KEYWORD", "RS": "RATIO", "R": "RATIO", "GOR": "RATIO"}
+            )
+
+        # Ensure that the identifier string "KEYWORD" is contained in the header columns
+        if "KEYWORD" not in self.pvt_df.columns:
+            raise ValueError(
+                (
+                    "There has to be a KEYWORD or TYPE column with corresponding Eclipse keyword."
+                    "When not providing a csv file, make sure ecl2df is installed."
+                )
+            )
+
+        self.phases_additional_info: List[str] = []
+        if self.pvt_df["KEYWORD"].str.contains("PVTO").any():
+            self.phases_additional_info.append("PVTO")
+        elif self.pvt_df["KEYWORD"].str.contains("PVDO").any():
+            self.phases_additional_info.append("PVDO")
+        elif self.pvt_df["KEYWORD"].str.contains("PVCDO").any():
+            self.phases_additional_info.append("PVCDO")
+        if self.pvt_df["KEYWORD"].str.contains("PVTG").any():
+            self.phases_additional_info.append("PVTG")
+        elif self.pvt_df["KEYWORD"].str.contains("PVDG").any():
+            self.phases_additional_info.append("PVDG")
+        if self.pvt_df["KEYWORD"].str.contains("PVTW").any():
+            self.phases_additional_info.append("PVTW")
+
+        self.add_store(
+            PluginIds.Stores.SELECTED_COLOR, WebvizPluginABC.StorageType.SESSION
+        )
+        self.add_store(
+            PluginIds.Stores.SELECTED_ENSEMBLES, WebvizPluginABC.StorageType.SESSION
+        )
+
+        self.add_store(
+            PluginIds.Stores.SELECTED_PHASE, WebvizPluginABC.StorageType.SESSION
+        )
+        self.add_store(
+            PluginIds.Stores.SELECTED_PVTNUM, WebvizPluginABC.StorageType.SESSION
+        )
+
+        self.add_store(
+            PluginIds.Stores.SELECTED_SHOW_PLOTS, WebvizPluginABC.StorageType.SESSION
+        )
```

Comment:
Stores should only be used when really necessary. Otherwise, they unnecessarily increase data traffic between client and server. Instead of taking the detour via stores, directly update the respective graphs/components in the callbacks.